docs/git/index.md

```diff
@@ -0,0 +1,220 @@
+
+## How does Git do that?
+
+Imagine your teacher wanted everyone in the class to answer three questions and to send back the answers. What capabilities would we need in order to be able to do that?
+
+**First**, the teacher needs to be able to write the questions, and **store** them somewhere.
+
+When we're using Git, we write things down in files in a folder (these could be text files, Word documents, images, or really any kind of file). When we save these files, we We say we're _committing_ them, and we call the folder where we're saving them a _repository_ (or _repo_ for short).
+
+TODO: Insert graphic of teacher writing and committing.
+
+**Next**, the teacher needs to put the repository somewhere where the learners can get it. We call this _pushing_ the repository. The teacher won't send a copy to each learner, but will put one copy somewhere on the Internet, and tell the learners where it is. The place the teachers on this course will be pushing to is a website called [Github](https://github.com), but there are other websites they could push to if they wanted to. This is the difference between Git and Github - Git is a way of storing and sharing files, and Github is one website where you can use Git.
+
+**Then**, now that the teacher has pushed the questions, each learner needs to be able to **get the questions onto their computer**.
+
+When we're using Git, we call this _cloning_ the teacher's repository (because we're making our own copy). After we've cloned the repository, we will have the same folder on our computer as the teacher created, committed, and pushed.
+
+TODO: Insert sequence diagram of teacher committing, pushing, and learners cloning.
+
+### Exercise (10 minutes)
+
+Code Your Future has already pushed an example repository, so you're going to try to clone it from Github onto your computer.
+
+1. Open up Github Desktop, and sign in if you're not already signed in.
+
+TODO: Insert screenshots for each step.
+
+2. You should see a button saying "Clone a Repository from the Internet" - click it. (If you don't see it, look in the menu at the top of the screen, open the "File" menu, and select the "Clone Repository" option).
+
+3. Now we need to tell Github Desktop where we want to repository we want to clone is. Choose the "URL" heading, and paste into the URL box: `https://github.com/illicitonion/cyf-demo-repo` (TODO: Replace with a CYF repo). The second box ("Local Path") is saying where it's going to clone it to on your computer. That's all Github Desktop is doing - copying a repository _from_ Github _to_ your computer. We don't need to change this box, but it can be useful to know where we can find our files later on.
+
+4. Press the Clone button. Github Desktop should soon show you a screen like this:
+
+TODO: Screenshot
+
+5. Click the "Open in Visual Studio Code" button (TODO: Explain to Linux users how to do this, because they probably don't have the button).
```

Comment:
From @40thieves:

Did you check if this is following `$EDITOR`? (My `$EDITOR` is set to VSCode and I see the same thing)

Reply:
It doesn't, it's specifically a VS Code integration